osbrain/agent.py

```diff
@@ -771,8 +771,8 @@ def _connect_address(self, server_address, alias=None, handler=None):
         assert server_address.role == 'server', \
             'Incorrect address! A server address must be provided!'
         client_address = server_address.twin()
-        assert not client_address.kind.requires_handler() or \
-            handler is not None, 'This socket requires a handler!'
+        assert not (client_address.kind.requires_handler() and not handler), \
+            'This socket requires a handler!'
```

Comment:
I actually prefer something in between:

    assert not client_address.kind.requires_handler() or handler

:joy: